doc/devel/contributing.rst

```diff
@@ -0,0 +1,422 @@
+i.e., it just passes all ``args`` and ``kwargs`` on to
+:meth:`matplotlib.text.Text.__init__`::
+
+  # in axes.py
+  def text(self, x, y, s, fontdict=None, withdash=False, **kwargs):
+      t = Text(x=x, y=y, text=s, **kwargs)
+
+and :meth:`~matplotlib.text.Text.__init__` (again with liberties for
+illustration) just passes them on to the
+:meth:`matplotlib.artist.Artist.update` method::
+
+  # in text.py
+  def __init__(self, x=0, y=0, text='', **kwargs):
+      Artist.__init__(self)
+      self.update(kwargs)
+
+``update`` does the work looking for methods named like
+``set_property`` if ``property`` is a keyword argument.  i.e., no one
+looks at the keywords, they just get passed through the API to the
+artist constructor which looks for suitably named methods and calls
+them with the value.
+
+As a general rule, the use of ``**kwargs`` should be reserved for
+pass-through keyword arguments, as in the example above.  If all the
+keyword args are to be used in the function, and not passed
+on, use the key/value keyword args in the function definition rather
+than the ``**kwargs`` idiom.
+
+In some cases, you may want to consume some keys in the local
+function, and let others pass through.  You can ``pop`` the ones to be
+used locally and pass on the rest.  For example, in
+:meth:`~matplotlib.axes.Axes.plot`, ``scalex`` and ``scaley`` are
+local arguments and the rest are passed on as
+:meth:`~matplotlib.lines.Line2D` keyword arguments::
+
+  # in axes.py
+  def plot(self, *args, **kwargs):
+      scalex = kwargs.pop('scalex', True)
+      scaley = kwargs.pop('scaley', True)
+      if not self._hold: self.cla()
+      lines = []
+      for line in self._get_lines(*args, **kwargs):
+          self.add_line(line)
+          lines.append(line)
+
+Note: there is a use case when ``kwargs`` are meant to be used locally
+in the function (not passed on), but you still need the ``**kwargs``
+idiom.  That is when you want to use ``*args`` to allow variable
+numbers of non-keyword args.  In this case, python will not allow you
+to use named keyword args after the ``*args`` usage, so you will be
+forced to use ``**kwargs``.  An example is
+:meth:`matplotlib.contour.ContourLabeler.clabel`::
+
+  # in contour.py
+  def clabel(self, *args, **kwargs):
+      fontsize = kwargs.get('fontsize', None)
+      inline = kwargs.get('inline', 1)
+      self.fmt = kwargs.get('fmt', '%1.3f')
+      colors = kwargs.get('colors', None)
+      if len(args) == 0:
+          levels = self.levels
+          indices = range(len(self.levels))
+      elif len(args) == 1:
+         ...etc...
+
+.. _custom_backend:
+
+Developing a new backend
+------------------------
+
+If you are working on a custom backend, the *backend* setting in
+:file:`matplotlibrc` (:ref:`customizing-matplotlib`) supports an
+external backend via the ``module`` directive.  if
+:file:`my_backend.py` is a matplotlib backend in your
+:envvar:`PYTHONPATH`, you can set use it on one of several ways
```

Comment:
delete "use"?  or delete "set"?
All of the following looks like it might be replaced with a link to somewhere else in the docs.